src/odbc/src/statement.cpp

```diff
@@ -994,6 +1007,73 @@ SqlResult::Type Statement::InternalClose() {
   return result;
 }
 
+void Statement::ExtendedFetch(SQLUSMALLINT orientation, SQLLEN offset, SQLULEN* rowCount, SQLUSMALLINT* rowStatusArray) {
+    IGNITE_ODBC_API_CALL(InternalExtendedFetch(orientation, offset, rowCount, rowStatusArray));
+}
+
+SqlResult::Type Statement::InternalExtendedFetch(SQLUSMALLINT orientation, SQLLEN offset,
+    SQLULEN* rowCount, SQLUSMALLINT* rowStatusArray) {
+  LOG_DEBUG_MSG("InternalExtendedFetch called with orientation " << orientation);
+  UNREFERENCED_PARAMETER(offset);
+
+  if (orientation != SQL_FETCH_NEXT) {
+    AddStatusRecord(SqlState::SHYC00_OPTIONAL_FEATURE_NOT_IMPLEMENTED,
+      "Only SQL_FETCH_NEXT FetchOrientation type is supported");
+    return SqlResult::AI_ERROR;
+  }
+
+  if (rowCount)
+    *rowCount = 0;
+
+  if (!currentQuery.get()) {
+    std::string errMsg = "Cursor is not in the open state.";
+    AddStatusRecord(SqlState::S24000_INVALID_CURSOR_STATE, errMsg);
+    return SqlResult::AI_ERROR;
+  }
+
+  SqlUlen columnBindOffsetValue = columnBindOffset ? *columnBindOffset : 0;
```

Comment:
This should not use columnBindOffset which comes from SQL_ATTR_BIND_OFFSET_PTR. It should explicitly set ByteOffset to zero since SQLExtendedFetch doesn't support bind offsets.

Reply:
Changed so that each byte offset is set to `0` within the loop just below this line.